virtual_rainforest/models/litter/litter_model.py

```diff
@@ -0,0 +1,178 @@
+        layer_roles = set_layer_roles(canopy_layers, soil_layers)
+        # Find first soil layer from the list of layer roles
+        self.top_soil_layer_index = next(
+            i for i, v in enumerate(layer_roles) if v == "soil"
+        )
+        """The layer in the data object representing the first soil layer."""
+        # Find first soil layer from the list of layer roles
+        self.surface_layer_index = next(
+            i for i, v in enumerate(layer_roles) if v == "surface"
+        )
+        """The layer in the data object representing the surface layer."""
+
+    @classmethod
+    def from_config(
+        cls, data: Data, config: dict[str, Any], update_interval: Quantity
+    ) -> LitterModel:
+        """Factory function to initialise the litter model from configuration.
+
+        This function unpacks the relevant information from the configuration file, and
+        then uses it to initialise the model. If any information from the config is
+        invalid rather than returning an initialised model instance an error is raised.
+
+        Args:
+            data: A :class:`~virtual_rainforest.core.data.Data` instance.
+            config: The complete (and validated) Virtual Rainforest configuration.
+            update_interval: Frequency with which all models are updated
+        """
+
+        # Find number of soil and canopy layers
+        soil_layers = config["core"]["layers"]["soil_layers"]
+        canopy_layers = config["core"]["layers"]["canopy_layers"]
+
+        # Check if any constants have been supplied
+        if "litter" in config and "constants" in config["litter"]:
+            # Checks that constants is config are as expected
+            check_valid_constant_names(config, "litter", "LitterConsts")
+            # If an error isn't raised then generate the dataclass
+            constants = LitterConsts(**config["litter"]["constants"]["LitterConsts"])
+        else:
+            # If no constants are supplied then the defaults should be used
+            constants = LitterConsts()
+
+        LOGGER.info(
+            "Information required to initialise the litter model successfully "
+            "extracted."
+        )
+        return cls(data, update_interval, soil_layers, canopy_layers, constants)
+
+    def setup(self) -> None:
+        """Placeholder function to setup up the litter model."""
+
+    def spinup(self) -> None:
+        """Placeholder function to spin up the litter model."""
+
+    def update(self, time_index: int) -> None:
+        """Calculate changes in the litter pools and use them to update the pools.
+
+        Args:
+            time_index: The index representing the current time step in the data object.
+        """
+
+        # Find all litter pools
+        litter_pools = {
+            str(name): self.data[str(name)].to_numpy()
+            for name in self.data.data.keys()
+            if str(name).startswith("litter_pool_")
+        }
```

Comment:
Why are you casting `name` to `str`? Aren't the keys always strings?

Reply:
If I don't cast them I get a `mypy` error because apparently `name` is of type `Hashable` when the dictionary expects the key to be of type `str`. There might be a smarter way around this though?

Reply:
I assume you've done it like this because the data being used by in the Litter Pools might expand. It does seem odd to make a kind of 'mini-Data' object though. I personally would probably explicitly unpack the expected variables into `calculate_litter_pool_updates` and skip this completely:

```python
updated_litter_pools = calculate_litter_pool_updates(
        surface_temp=self.data["air_temperature"][
            self.surface_layer_index
        ].to_numpy(),
        constants=self.constants,
        update_interval=self.update_interval.to("day").magnitude,
    litter_pool_above_metabolic= self.data['litter_pool_above_metabolic'].to_numpy(),
    litter_pool_above_structural=self.data['litter_pool_above_metabolic'].to_numpy(),
)
```

Alternatively, you could pass in `data = self.data` to keep the API simpler, but I think an explicit list of inputs is cleaner.

Reply:
I mean it's not like the function is going to do anything with `data['litter_pools_fiz_buz`]` apart from raise `TypeError: calculate_litter_pool_updates() got an unexpected keyword argument`

Reply:
Yeah this is copied over from the `soil` module where there's more complex ordering of pools going on (i.e. into a 1D `numpy` array and back again). Kept it in because it saved a small amount of effort, but that's not really a good enough reason to keep it